### PR TITLE
cluster-ui: add `safesql` to `cluster-ui`

### PR DESCRIPTION
**Pull request by THardy98**
Epic: none

This change ports the `safesql` package from managed-service to cluster-ui. Safesql ensures that SQL literals/identifiers are properly quoted.

Release note: None

### CODE REVIEW

**Review by matthewtodd — COMMENTED** [review]
Hey, @THardy98! Just wanted to double-check: IIUC, this library does (and should) stop short of interpolating arguments to queries, yes? It's strictly for things like user, database, table, and column names? And everything else we'd pass along for the db driver to interpolate -- I think drawing that line limits our security exposure.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-:-NNCLAGv1Vntxur7pYo0:bsmk00f)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by THardy98 — COMMENTED** [review]
I believe so yes. The most common use case I've seen being adding identifiers directly to query strings:
```
safesql.Format(`SELECT * FROM %1.this_is_a_table_name`, safesql.Identifier(`this is my db name`))
```

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-:-NNIZpRn6PUQGwklD7SY:b-qvr7nf)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by maryliag — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96286)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @matthewtodd and @THardy98)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/safesql.ts` line 22 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-NNbA_Zr6r7O-jgy1BcE:-NNbA_Zr6r7O-jgy1BcF:bytujyy) ([raw file](https://github.com/cockroachdb/cockroach/blob/1403411188fa938d300f316793f781e2f3d4618a/pkg/ui/workspaces/cluster-ui/src/api/safesql.ts#L22)):*
> ```XML
> }
> 
> // TODO(thomas): @knz says: the quoting rules for usernames are different from
> ```

are you addressing this on this PR?

**Review by THardy98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96286)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @maryliag and @matthewtodd)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/safesql.ts` line 22 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-NNbA_Zr6r7O-jgy1BcE:-NNbH9pd-WG-TqB6n7WY:bfegyvo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1403411188fa938d300f316793f781e2f3d4618a/pkg/ui/workspaces/cluster-ui/src/api/safesql.ts#L22)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

are you addressing this on this PR?
</blockquote></details>

TLDR: no

This was actually an old `TODO` left by Peter a couple years ago (late 2020) on the corresponding `safesql.go` in managed-service. I decided to keep it and tag myself here as I've added this. Alternatively, I could omit the `TODO` altogether.

**Review by maryliag — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-:-NNbKmoq0SHb5tEa27AD:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @matthewtodd and @THardy98)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/safesql.ts` line 22 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96286#-NNbA_Zr6r7O-jgy1BcE:-NNbKjHA-mf0YF79wVdq:bglc2or) ([raw file](https://github.com/cockroachdb/cockroach/blob/1403411188fa938d300f316793f781e2f3d4618a/pkg/ui/workspaces/cluster-ui/src/api/safesql.ts#L22)):*
<details><summary><i>Previously, THardy98 (Thomas Hardy) wrote…</i></summary><blockquote>

TLDR: no

This was actually an old `TODO` left by Peter a couple years ago (late 2020) on the corresponding `safesql.go` in managed-service. I decided to keep it and tag myself here as I've added this. Alternatively, I could omit the `TODO` altogether.
</blockquote></details>

You can keep it, np!

**Comment by THardy98** [comment]
TYFR :)
